apps/examples/testcase/le_tc/kernel/tc_libc_pthread.c

```diff
@@ -1227,6 +1230,85 @@ static void tc_libc_pthread_pthread_rwlock_trywrlock(void)
 	TC_SUCCESS_RESULT();
 }
 
+/**
+* @fn                   :tc_libc_pthread_pthread_setcancelstate
+* @brief                :This tc tests pthread_setcancelstate()
+* @Scenario             :The function shall atomically both set the calling thread's cancelability state to the indicated state 
+*                        and return the previous cancelability state at the location referenced by oldstate
+*                        If successful pthread_setcancelstate() function shall return zero;
+*                        otherwise, an error number shall be returned to indicate the error.
+* @API'scovered         :pthread_setcancelstate
+* @Preconditions        :none
+* @Postconditions       :none
+* @return               :void
+*/
+static void tc_libc_pthread_pthread_setcancelstate(void)
```

Comment:
I think second is better. TC was already split by location. If you do first, it makes location of TC wrong.